src/Util/StreamInfo.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Util;
+
+/**
+ * A utility class to get various information from a file stream.
+ */
+class StreamInfo
+{
+    /**
+     * Get the hash for file stream.
+     *
+     * @param array $fileStream The file stream array.
+     *
+     * @return string The hash calculated from the stream.
+     */
+    public static function calculateHash(array $inputFileStream, string $algo = 'sha256') :string
+    {
+        $fileStream = $inputFileStream['fileStream'] ?? null;
+        $context = hash_init($algo);
+        hash_update_stream($context, $fileStream);
```

Comment:
This would fail if $fileStream is null on type error

Reply:
Yes, and that's intentionally. I don't want to catch any errors here, because the caller does a try catch anyway, so it's the same scenario if we would call the native functions.